File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n=nums.length;
        int start=0,sum=0;
        int length=Integer.MAX_VALUE;
        for(int i=0;i<n;i++){
            sum=sum+nums[i];
            while(sum>=target){
                length=Math.min(length,i-start+1);
                sum=sum-nums[start];
                start++;
            }
        }
        return length==Integer.MAX_VALUE?0:length;
    }
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n=len(nums)
        left=0
        total=0
        length=float('inf')
        for right in range(n):
            total=total+nums[right]
            while total>=target:
                length=min(length,right-left+1)
                total=total-nums[left]
                left=left+1
        
        if(length==float(inf)):
            length=0

        return length

        